rest/src/main/kotlin/builder/webhook/EditWebhookMessageBuilder.kt

```diff
@@ -0,0 +1,35 @@
+package dev.kord.rest.builder.webhook
+
+import dev.kord.common.entity.AllowedMentions
+import dev.kord.common.entity.optional.Optional
+import dev.kord.common.entity.optional.delegate.delegate
+import dev.kord.rest.builder.RequestBuilder
+import dev.kord.rest.builder.message.EmbedBuilder
+import dev.kord.rest.json.request.EmbedRequest
+import dev.kord.rest.json.request.WebhookEditMessageRequest
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+class EditWebhookMessageBuilder : RequestBuilder<WebhookEditMessageRequest> {
+
+    private var _content: Optional<String> = Optional.Missing()
+    var content: String? by ::_content.delegate()
+
+    var embeds: MutableList<EmbedRequest> = mutableListOf()
+
+    private var _allowedMentions: Optional<AllowedMentions> = Optional.Missing()
+    var allowedMentions: AllowedMentions? by ::_allowedMentions.delegate()
+
+    @OptIn(ExperimentalContracts::class)
+    inline fun embed(builder: EmbedBuilder.() -> Unit) {
+        contract {
+            callsInPlace(builder, InvocationKind.EXACTLY_ONCE)
+        }
+        embeds.add(EmbedBuilder().apply(builder).toRequest())
```

Comment:
Make the embeds a list of EmbedBuilder, and transform it to a request down in the toRequest. this will allow the user to modify the list after done adding or perform whatever they please